### PR TITLE
Add missing sounds when equipping or unequipping armour using right-c…

### PR DESCRIPTION
**Pull request by zSALLAZAR**
…lick

### Relevant issues

#5167 

## Tests

https://user-images.githubusercontent.com/59490940/202156971-89b2d3e7-b24a-4711-b335-b52f18e415cd.mp4

### CODE REVIEW

**Comment by NhanAZ** [comment]
Looks like the sound of Netherite armor is missing

**Comment by zSALLAZAR** [comment]
> Looks like the sound of Netherite armor is missing

Yeah, but Netherite is not implemented in stable

**Comment by NhanAZ** [comment]
I didn't look at the branch 💀 
I think you should move this pull request to branch `next-major`
https://github.com/pmmp/PocketMine-MP/blob/stable/CONTRIBUTING.md#choosing-a-target-branch

**Comment by zSALLAZAR** [comment]
You sure? I think I should change the targeting branch to next-minor because I only added new API classes

**Comment by dktapps** [comment]
next-minor is fine

File: src/item/Armor.php [hunk 1]
```diff
@@ -125,6 +130,18 @@ public function onClickAir(Player $player, Vector3 $directionVector) : ItemUseRe
...
 		$new = $thisCopy->pop();
 		$player->getArmorInventory()->setItem($this->getArmorSlot(), $new);
+		$sound = match($this->getId()){
+			ItemIds::TURTLE_HELMET, ItemIds::LEATHER_HELMET, ItemIds::LEATHER_CHESTPLATE, ItemIds::LEATHER_LEGGINGS, ItemIds::LEATHER_BOOTS => new ArmorEquipLeatherSound(),
+			ItemIds::CHAIN_HELMET, ItemIds::CHAIN_CHESTPLATE, ItemIds::CHAIN_LEGGINGS, ItemIds::CHAIN_BOOTS => new ArmorEquipChainSound(),
+			ItemIds::IRON_HELMET, ItemIds::IRON_CHESTPLATE, ItemIds::IRON_LEGGINGS, ItemIds::IRON_BOOTS => new ArmorEquipIronSound(),
+			ItemIds::DIAMOND_HELMET, ItemIds::DIAMOND_CHESTPLATE, ItemIds::DIAMOND_LEGGINGS, ItemIds::DIAMOND_BOOTS => new ArmorEquipDiamondSound(),
+			ItemIds::GOLD_HELMET, ItemIds::GOLD_CHESTPLATE, ItemIds::GOLD_LEGGINGS, ItemIds::GOLD_BOOTS => new ArmorEquipGoldSound(),
+			//TODO: Netherite equip sounds
+			default => null
```
Comment (dktapps) [inline src/item/Armor.php]: This is an awful mess. There has to be a better way to do this.

**Review by dktapps — CHANGES_REQUESTED** [review]

**Review by jasonw4331 — APPROVED** [review]

File: src/item/Armor.php [hunk 2]
```diff
@@ -69,6 +73,18 @@ public function getMaxStackSize() : int{
...
 	}
 
+	public function getEquipSound() : ?Sound{
+		return $this->equipSound;
+	}
+
+	/**
+	 * @return $this
+	 */
+	public function setEquipSound(?Sound $equipSound) : self{
```
Comment (dktapps) [inline src/item/Armor.php]: This should be removed. We don't persist this information, so it's going to vanish after a restart.

**Review by dktapps — APPROVED** [review]

**Comment by ShockedPlot7560** [comment]
Is merge planned?

**Comment by dktapps** [comment]
Yes, but I'll need to re-review it once the branches are in sync

**Comment by zSALLAZAR** [comment]
I think it would be better if we add the equip sounds to `src/item/ArmorMaterial.php` to prevent duplication, as also mentioned in https://github.com/pmmp/PocketMine-MP/pull/5953#issue-1829965310

**Comment by dktapps** [comment]
Yeah, I agree

**Comment by ShockedPlot7560** [comment]
Superseeded by https://github.com/pmmp/PocketMine-MP/commit/f799cfaba6318dcf90fce3956bb4843d12581df2

**Pull request closed without merging**